### PR TITLE
Feat: add builder methods to exp.Update and add with_ arg to exp.update

### PR DESCRIPTION
**Pull request by brdbry**
Improve ergonomics of UPDATEs:

- Add builder methods to Update class so it can be constructed incrementally in the same way a Select
- `exp.update` changes:
  - Add a `with_` arg so updates with CTEs can be created in one-shot without subsequent need  to `.set(...)` the With clause
  - Make the `properties` arg optional, so `set_` builder method can be used with an AST (`exp.update("tbl").set_(set_expr)` ) instead of forcing specification as a `dict`

### CODE REVIEW

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -3284,6 +3284,191 @@ class Update(Expression):
...
+                If nothing is passed in then a from is not applied to the expression
+            dialect: the dialect used to parse the input expression.
+            copy: if `False`, modify this expression instance in-place.
+            opts: other options to use to parse the input expressions.
+
+        Returns:
+            The modified Update expression.
+        """
+        if not expression:
+            return self if not copy else self.copy()
```
Comment (georgesittas) [inline sqlglot/expressions.py:3413]:
```suggestion
            return maybe_copy(self)
```

File: sqlglot/expressions.py [hunk 2]
```diff
@@ -3284,6 +3284,191 @@ class Update(Expression):
...
+            into=Table,
+            prefix=None,
+            dialect=dialect,
+            copy=copy,
+            **opts,
+        )
+
+    def set_(
+        self, *expressions: ExpOrStr, dialect: DialectType = None, copy: bool = True, **opts
+    ) -> Update:
```
Comment (georgesittas) [inline sqlglot/expressions.py:3321]: We probably want an "append" mode like we do for, e.g., `select`. Right now you always overwrite which isn't as flexible.

File: sqlglot/expressions.py [hunk 3]
```diff
@@ -6803,38 +6988,41 @@ def from_(expression: ExpOrStr, dialect: DialectType = None, **opts) -> Select:
 
 def update(
     table: str | Table,
-    properties: dict,
+    properties: t.Optional[dict] = None,
```
Comment (georgesittas) [inline sqlglot/expressions.py:6991]: Why did this become optional?

File: sqlglot/expressions.py [hunk 4]
```diff
@@ -6847,6 +7035,15 @@ def update(
             "where",
             maybe_parse(where, into=Where, dialect=dialect, prefix="WHERE", **opts),
         )
+    if with_:
+        cte_list = [
+            CTE(this=maybe_parse(qry, into=Select, dialect=dialect, **opts), alias=alias)
```
Comment (georgesittas) [inline sqlglot/expressions.py:7040]: Will this work for set operators as well? For example `UNION`, etc.

**Review by georgesittas — COMMENTED** [review]
LGTM, just a few comments. Thanks a lot for the PR, @brdbry!

File: sqlglot/expressions.py (hunk 2 above)
Comment (brdbry) [reply sqlglot/expressions.py]: Good idea, done

File: sqlglot/expressions.py (hunk 3 above)
Comment (brdbry) [reply sqlglot/expressions.py:7000]: I put it in the PR comments at the top. Making it mandatory prevents you from using the exp.update function when you already have a list SET expressions, so it's not consistent with other helper methods like exp.select

Comment (georgesittas) [reply sqlglot/expressions.py:7000]: Ah, I see; apologies, I didn't notice the description. Will take another look.

**Review by georgesittas — APPROVED** [review]

File: sqlglot/expressions.py (hunk 3 above)
Comment (georgesittas) [reply sqlglot/expressions.py:7000]: Sounds reasonable 👍

**Pull request merged**